rasn-compiler/src/lexer/mod.rs

```
@@ -231,6 +232,32 @@ fn top_level_value_declaration(input: Input<'_>) -> ParserResult<'_, ToplevelVal
     ))(input)
 }
 
+fn top_level_valueset_declaration(input: Input<'_>) -> ParserResult<'_, ToplevelTypeDefinition> {
+    map(
+        tuple((
+            skip_ws(many0(comment)),
+            skip_ws(context_boundary(title_case_identifier)),
+            skip_ws_and_comments(opt(parameterization)),
+            skip_ws_and_comments(asn1_type),
+            preceded(assignment, in_braces(element_set))
+        )),
+        |mut value| {
+            if let Some(constraints) = value.3.constraints_mut() {
+                constraints.push(Constraint::SubtypeConstraint(value.4))
+            } else {
+                eprintln!("{name}: unable to push constraints to {ty:?}", name=value.1, ty=value.3);
```

Could you return an error here instead of logging, so that we report errors consistently through the Result type?